java/code/src/com/redhat/rhn/manager/system/entitling/SystemUnentitler.java

```diff
@@ -40,8 +41,17 @@
 
     private static final Logger LOG = Logger.getLogger(SystemUnentitler.class);
 
-    public static final SystemUnentitler INSTANCE = new SystemUnentitler();
     private MonitoringManager monitoringManager = new FormulaMonitoringManager();
+    private VirtManager virtManager;
+
+    /**
+     * @param virtManagerIn instance for managing virtual machines.
+     * @param monitoringManagerIn instance for handling monitoring configuration.
+     */
+    public SystemUnentitler(VirtManager virtManagerIn, MonitoringManager monitoringManagerIn) {
+        this.virtManager = virtManagerIn;
+        this.monitoringManager = monitoringManagerIn;
```

Comment:
Thanks for setting `monitoringManager` here in the constructor now, maybe we no longer need to initialize it above?

Reply:
right

Reply:
done